apps/api/src/user/service/user.service.ts

```diff
@@ -172,4 +175,38 @@ export class UserService {
 
     return newUser
   }
+
+  private async checkIfAdminExistsOrCreate() {
+    const adminExists =
+      (await this.prisma.user.count({
+        where: {
+          isAdmin: true
+        }
+      })) > 0
+
+    if (!adminExists) {
+      this.log.warn('No admin user found', 'UserService')
+      await this.createAdminUser()
+      return
+    }
+    this.log.log('Admin user found', 'UserService')
+  }
+
+  private async createAdminUser() {
```

Comment:
You can merge these two functions i would say. we wont be using it any place else